SpinCore_pp/save_data.py

```diff
@@ -0,0 +1,76 @@
+import h5py
+import os
+import pyspecdata as psd
+import pyspecProcScripts
+import subprocess
+
+
+def save_data(dataset, target_directory, config_dict, counter_type):
+    """save data to an h5 file with appropriately labeled nodename and performs
+    rough processing
+
+    Parameters
+    ==========
+    dataset: nddata
+        acquired data in nddata format
+    my_exp_type: str
+        directory on the share drive you want to save to
+    config_dict: dict
+        config_dict pulled from the active.ini file
+    counter_type: str
+        type of counter you are incrementing
+
+    Returns
+    =======
+    config_dict: dict
+        the updated config dict after appropriately incrementing the counter
+    """
+    # {{{ create filename
+    filename = f"{config_dict['date']}_{config_dict['chemical']}_{config_dict['type']}"
+    # }}}
+    nodename = config_dict["type"] + "_" + str(config_dict["%s_counter" % counter_type])
+    dataset.name(nodename)
+    filename_out = filename + ".h5"
+    if os.path.exists(f"{target_directory}{filename_out}"):
+        print("this file already exists so we will add a node to it!")
+        with h5py.File(
+            os.path.normpath(os.path.join(target_directory, f"{filename_out}"))
+        ) as fp:
+            while nodename in fp.keys():
+                config_dict["%s_counter" % counter_type] += 1
+                nodename = (
+                    config_dict["type"]
+                    + "_"
+                    + str(config_dict["%s_counter" % counter_type])
+                )
+            dataset.name(nodename)
+    dataset.hdf5_write(f"{filename_out}", directory=target_directory)
+    print("\n** FILE SAVED IN TARGET DIRECTORY ***\n")
+    target_directory_parts = target_directory.split("\\")
+    my_exp_type = "/".join((target_directory_parts[-3], target_directory_parts[-2]))
```

Comment:
this line is really problematic, because it's not going to work in general.  I think we have two options:
* yes, because of this go back to using exp_type and converting it to a path here (*as well as* at the top of the file where you are converting to a path now)
* passing both the path and the exp type to here

Reply:
so, to be clear, I'm stepping back what I said earlier -- I didn't know that we *needed* exp_type for this function.

By "convert to path" I mean via `getDATADIR`

I'm not walking back, however, the idea that you *do* want to check that the directory exists at the *top* of your ppg, because you don't want to run the whole thing and then get weird problems b/c the dir doesn't exist

Reply:
I ran it and it works but I implemented a different version and again ran the examples and it works so what is there now should be good unless you still don't like the format

Reply:
objectively what is there is not good